src/operator/swapaxis-inl.h

```diff
@@ -135,7 +135,7 @@ class SwapAxisOp : public Operator {
                        const std::vector<TBlob> &aux_args) {
     using namespace mshadow;
     Stream<xpu> *s = ctx.get_stream<xpu>();
-
+    CHECK_NE(req[swapaxisenum::kOut], kAddTo);
```

Comment:
it would probably make sense to add a message here to not get very cryptic error messages. Would it not be possible to apply what has been done for the backward pass?